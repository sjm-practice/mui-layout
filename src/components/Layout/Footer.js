import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/styles";
import useConfig from "../../hooks/useConfig";

const useStyles = makeStyles(({ breakpoints, palette, spacing, transitions }) => ({
  root: {
    borderTop: "1px solid",
    borderColor: palette.grey[200],
    padding: spacing(2),
    [breakpoints.up("sm")]: {
      padding: spacing(3),
    },
    transition: transitions.create(["margin"], {
      easing: transitions.easing.sharp,
      duration: transitions.duration.leavingScreen,
    }),
  },
}));

const Footer = ({ className, component: Component, style, ...props }) => {
  const ctx = useConfig();
  const {
    navVariant,
    navWidth,
    collapsible,
    collapsed,
    collapsedWidth,
    footerShrink,
    open,
    navAnchor,
  } = ctx;

  const getMargin = () => {
    if (navAnchor !== "left" || !footerShrink) return 0;
    if (navVariant === "persistent" && open) {
      // open is effect only when
      // navVariant === 'persistent' ||
      // navVariant === 'temporary'
      return navWidth;
    }
    if (navVariant === "permanent") {
      if (collapsible) {
        if (collapsed) return collapsedWidth;
        return navWidth;
      }
      return navWidth;
    }
    return 0;
  };

  const classes = useStyles(props);

  return (
    <Component
      {...props}
      className={`${classes.root} ${className}`}
      style={{
        ...style,
        marginLeft: getMargin(),
      }}
    />
  );
};

Footer.propTypes = {
  className: PropTypes.string,
  component: PropTypes.elementType,
  style: PropTypes.shape({}),
};
Footer.defaultProps = {
  className: "",
  component: "footer",
  style: {},
};

export default Footer;
