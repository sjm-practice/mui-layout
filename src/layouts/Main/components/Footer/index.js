import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { makeStyles } from "@material-ui/styles";
import { Typography, Link } from "@material-ui/core";

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(4),
  },
}));

const Footer = props => {
  const { className, ...rest } = props;

  const classes = useStyles();

  return (
    <div {...rest} className={clsx(classes.root, className)}>
      <Typography variant="body1">
        &copy;{" "}
        <Link component="a" href="https://avoinsights.com" target="_blank">
          AVO Insights
        </Link>
        . 2019
      </Typography>
      <Typography variant="caption">
        From big initiatives to our SaaS product, we create useful Augmented Reality...
      </Typography>
    </div>
  );
};

Footer.propTypes = {
  className: PropTypes.string,
};

Footer.defaultProps = {
  className: null,
};

export default Footer;
