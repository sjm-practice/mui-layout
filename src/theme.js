import { createMuiTheme } from "@material-ui/core/styles";
import { red } from "@material-ui/core/colors";

// Create a theme instance.
const theme = createMuiTheme({
  palette: {
    error: {
      main: red.A400,
    },
  },
});

export default theme;
