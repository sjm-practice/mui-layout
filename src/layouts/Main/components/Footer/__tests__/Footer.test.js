/* eslint-env jest */

import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/styles";
import theme from "../../../../../theme";

import Footer from "../index";

describe("<Footer />", () => {
  it("Should match Snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Footer />
      </ThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
