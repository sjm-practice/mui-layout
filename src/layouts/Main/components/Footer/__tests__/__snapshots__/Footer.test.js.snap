// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Â© 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://avoinsights.com"
        target="_blank"
      >
        AVO Insights
      </a>
      . 2019
    </p>
    <span
      class="MuiTypography-root MuiTypography-caption"
    >
      From big initiatives to our SaaS product, we create useful Augmented Reality...
    </span>
  </div>
</DocumentFragment>
`;
